src/postponed_job.rs

```diff
@@ -0,0 +1,76 @@
+//! Functions for interacting with the postponed jobs API.
+
+use crate::Ruby;
+
+use rb_sys::rb_postponed_job_register;
+use std::ffi::c_void;
+
+impl Ruby {
+    /// Registers a function to be executed when the Ruby VM is safe to access
+    /// (i.e. inside a signal handler or during a GC run).
+    ///
+    /// ```rust
+    /// # let _cleanup = unsafe { magnus::embed::init() };
+    ///
+    /// use magnus::{Ruby, Value};
+    /// use magnus::postponed_job::RegistrationError;
+    /// use std::sync::atomic::{AtomicUsize, Ordering};
+    ///
+    /// static COUNTER: AtomicUsize = AtomicUsize::new(0);
+    ///
+    /// let increment_counter = || {
+    ///   assert!(Ruby::get().is_ok());
+    ///
+    ///   COUNTER.fetch_add(1, Ordering::SeqCst);
+    /// };
+    ///
+    /// match Ruby::postponed_job_register(increment_counter) {
+    ///     Err(RegistrationError::BufferFull) => println!("Postponed job queue is full!"),
+    ///     Err(e) => println!("Error registering postponed job: {}", e),
+    ///     Ok(()) => println!("Postponed job registered!"),
+    /// };
+    ///
+    /// assert_eq!(0, COUNTER.load(Ordering::SeqCst));
+    /// let _: Value = magnus::eval!("sleep 0.1").unwrap();
+    /// assert_eq!(1, COUNTER.load(Ordering::SeqCst));
+    /// ```
+    pub fn postponed_job_register<F: FnOnce()>(func: F) -> Result<(), RegistrationError> {
+        let flags = 0;
+        let trampoline = trampoline::<F> as unsafe extern "C" fn(*mut c_void);
+        let boxed_func = Box::new(func);
+        let boxed_func_ptr = Box::into_raw(boxed_func) as *mut c_void;
+        let ret_val = unsafe { rb_postponed_job_register(flags, Some(trampoline), boxed_func_ptr) };
+
+        match ret_val {
+            0 => Err(RegistrationError::BufferFull),
+            1 => Ok(()),
+            v => unreachable!("rb_postponed_job_register returned an unknown value: {}", v),
```

Comment:
FWIW, in the case of `rb_postponed_job_register_one`, there's the [PBJJ_INTERRUPTED](https://github.com/ruby/ruby/blob/e135a21a85f658cea380df47dca16f5e90c971cf/vm_trace.c#L1665) case when there's already a job registered. 

It might not be important information to relay back to the user, but they did decide to return it for some reason.

Reply:
It looks like `PBJJ_INTERRUPTED` is an internal value that won't ever be returned from a public api.